qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -149,60 +60,104 @@ def settings(self) -> Dict[str, Any]:
+        result.fun = self._state.fun(self._state.x)
+        result.nfev = self._state.nfev
+        result.njev = self._state.njev
+        return result
+
+    def initialize(
         self,
         fun: Callable[[POINT], float],
         x0: POINT,
         jac: Optional[Callable[[POINT], POINT]] = None,
         bounds: Optional[List[Tuple[float, float]]] = None,
-    ) -> OptimizerResult:
-        # set learning rate
+    ) -> None:
+        """
+        This method will initialize the state of the optimizer so that an optimization can be performed.
+        It will always setup the initial point and will restart the counter for function evaluations.
+        This method is left blank because every optimizer has a different kind of state.
+        """
         if isinstance(self.learning_rate, float):
             eta = constant(self.learning_rate)
         else:
             eta = self.learning_rate()
 
-        if jac is None:
-            eps = 0.01 if self.perturbation is None else self.perturbation
-            jac = partial(
-                Optimizer.gradient_num_diff,
-                f=fun,
-                epsilon=eps,
-                max_evals_grouped=self._max_evals_grouped,
-            )
-
-        # prepare some initials
-        x = np.asarray(x0)
-        nfevs = 0
-
-        for _ in range(1, self.maxiter + 1):
-            # compute update -- gradient evaluation counts as one function evaluation
-            update = jac(x)
-            nfevs += 1
-
-            # compute next parameter value
-            x_next = x - next(eta) * update
-
-            # send information to callback
-            stepsize = np.linalg.norm(update)
-            if self.callback is not None:
-                self.callback(nfevs, x_next, fun(x_next), stepsize)
-
-            # update parameters
-            x = x_next
-
-            # check termination
-            if stepsize < self.tol:
-                break
+        self._state = GD_OptimizerState(
+            fun=fun,
+            jac=jac,
+            x=np.asarray(x0),
+            nit=0,
+            nfev=0,
+            njev=0,
+            eta=eta,
+            stepsize=np.inf,
+        )
 
-        # TODO the optimizer result should contain the number of gradient evaluations,
-        # if the gradient is passed
-        result = OptimizerResult()
-        result.x = x
-        result.fun = fun(x)
-        result.nfev = nfevs
+    def continue_condition(self) -> bool:
+        """
+        Condition that indicates the optimization process should come to an end.
+        When the stepsize is smaller than the tolerance, the optimization process is considered
+        finished.
+        Returns:
+            True if the optimization process should continue, False otherwise.
+        """
 
-        return result
+        cont_condition = self._state.stepsize > self.tol
+        cont_condition &= super().continue_condition()
+        return cont_condition
```

Comment:
Just a suggestion you're welcome to ignore, but maybe this would be more concise? 
```suggestion
        return cont_condition and super().continue_condition()
```